src/cms-driver.ts

```diff
@@ -197,15 +201,22 @@ export class CMakeServerClientDriver extends CMakeDriver {
 
   get cmakeCacheEntries(): Map<string, CacheEntryProperties> { return this._cacheEntries; }
 
+
   async doSetKit(need_clean: boolean, cb: () => Promise<void>): Promise<void> {
+    this._clientChangeInProgress = new Promise(async resolve => {
```

Comment:
This is dangerous (the promise can go unresolved). Here's my suggested refactor:

1. Move the inner body of the promise callback to a separate private function (Name something like `_setKitAndRestart`).
2. Have `doSetKit()` simply do `return this._clientChangeInProgress = this._setKitAndRestart(needClean, cb);`

The `_clientChangeInProgress` will be immediately set to the `Promise` for the inner function, and it will properly handle exceptions and resolve itself when `_setKitAndRestart` returns.

Reply:
in this case I had the problem in the execution order the promises. the used promise is used to block handling an driver call until client change is done.

I will refactor it later.

Reply:
Done